pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -56,8 +57,13 @@ type APIClient struct {
 
 // GetAPIClient returns the shared ApiClient instance.
 func GetAPIClient() (*APIClient, error) {
+	clientBuilder, err := NewClientBuilder()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This is not included in the retry logic since this has to do with creating the `rest.Config`. It seems appropriate to not retry if we can't even do that?